FHEM/00_SIGNALduino.pm

```diff
@@ -167,7 +167,7 @@ my %matchListSIGNALduino = (
 			"4:OREGON"						=> "^(3[8-9A-F]|[4-6][0-9A-F]|7[0-8]).*",
 			"7:Hideki"						=> "^P12#75[A-F0-9]+",
 			"9:CUL_FHTTK"					=> "^T[A-F0-9]{8}",
-			"10:SD_WS07"					=> "^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2}(#R[A-F0-9][A-F0-9]){0,1}\$",
+			"10:SD_WS07"					=> "^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2}+",
```

Comment:
Was genau soll das + am Ende der Regex bewirken? Verstehe ich leider nicht. :(
Fehler oder gewollt?

Reply:
Das Plus am Ende wird in %matchListSIGNALduino in 11 von 27 regex verwendet. Die Aussagen, wofür es steht, sind verschieden:
- Match 1 or more times
- The regex /-.+-/ then would match the same things as /-.*-/ 
- https://regex101.com/r/AtdlHq/3 - {2}+ Quantifier — Matches exactly 2 times

Besser wäre wahrscheinlich doch .*
Führt aber eigentlich zum selben Ergebnis: https://regex101.com/r/AtdlHq/4

Wann wird eigentlich die %matchListSIGNALduino in 00_SIGNALduino.pm überhaupt verwendet?
Die Nachrichten werden ja auch dispatcht, wenn dort etwas nicht passendes drin steht.

Reply:
Prinzipiell ist mir schon klar wofür das Plus in einer Regex steht.

Nur die Angabe {2} und + 
Also es soll genau zwei mal vorkommen
Es soll mindestens ein mal vorkommen oder belibig öfters.

.* Passt auch bei keinem vorkommen

Bezogen auf die Regex die ich angemerkt habe kann {2} durchaus korrekt sein.
Wenn kein $ am Ende der Regex steht, dann kann die Nachricht anschließend belibige Zeichen enthalten.

Den Zweck der Matchlist ist in folgendem Beitrag hinterlegt:

https://r.tapatalk.com/shareLink?url=https%3A%2F%2Fforum%2Efhem%2Ede%2Findex%2Ephp%3Ftopic%3D41994%2E0&share_tid=41994&share_fid=75100&share_type=t

Reply:
Danke für den Link, jetzt habe ich den Ablauf verstanden.
Wenn wir jetzt genau prüfen wollen, bräuchten wir folgende Regex https://regex101.com/r/nScALi/1 in der SIGNALduino **und** im SD_WS07:

^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2,3}(#R[A-F0-9]{1,2}){0,1}$

P7#,6x Nibble, 1x Nibble (AFaf), 2 oder 3 Nibble, evtl. #R (RSSI) mit ein oder 2 Nibble

Reply:
Denkst Du, es reicht nicht aus, nur auf den Anfang zu prüfen?

`^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2,3}`

Reply:
Langsam aber sicher machst du mich fertig bei der Hitze:-)
Das hatte ich doch bereits, dann kam dein Einwand vor 23h "'$' wäre das Ende der Zeichenkette und war bestimmt beabsichtigt"!
Also einigen wir uns jetzt auf:
^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2,3}
ohne + oder * am Ende???

Reply:
Mir ist auch war, aber da war doch ein \$ und kein $ am Ende?

So würde alles passen, was diesen Fang hat, was am Ende noch angehängt wird spielt keine Rolle:
`^P7#[A-Fa-f0-9]{6}[AFaf][A-Fa-f0-9]{2,3}`

Reply:
OK, dann machen wir es so.